hashicorp/boundary-ui
# Icu-11111 don't display empty modal when a target has no target address and no host sources

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-11111 )

## Description


If a target is missing an address and has no host sources then an empty modal was displayed. This is a bug. To fix this, the connect buttons were disabled and a new app state message was added for this case.

## Screenshots (if appropriate):
Before:
<img width="648" alt="Screenshot 2023-09-19 at 1 52 37 PM" src="https://github.com/hashicorp/boundary-ui/assets/107949262/b9d3bb47-e906-4b64-9db2-782bed284530">

After:
<img width="1013" alt="Screenshot 2023-09-19 at 3 36 49 PM" src="https://github.com/hashicorp/boundary-ui/assets/107949262/41a9446f-f335-4222-9fc3-2275ecdee705">

<img width="997" alt="Screenshot 2023-09-20 at 1 18 40 PM" src="https://github.com/hashicorp/boundary-ui/assets/107949262/921ae3ab-19c4-4d8e-8b16-76504d4f2e0b">

## How to Test
1. Run boundary in dev mode (use latest binary).
2. Create a target in admin UI with no hosts and no address.
3. Run desktop client using `ENABLE_MIRAGE=false yarn start:desktop` and while on this branch.
4. Open desktop client and validate that button in target list view is disabled and button in details view is disabled. Validate correct message is displayed. 


## Checklist:
